### ISSUE TITLE
прекоммент не у цита, а у форм? или посткоммент?

### ISSUE THREAD

**Issue by solanum-tuberosum** [obelyaev/abaevdict#91]
![image](https://user-images.githubusercontent.com/13102259/75093433-c06a4e80-5592-11ea-9e30-bb5daf4d3a04.png)
Слово kereke.
1. кирг. написала в прекоммент к циту, это так и надо?
2. вряд ли же "анат" относится к плащу? Наверно, это анат. панцирь? Там опечатка и это посткоммент к форме? Но посткоммент вылезает между словом и переводом... А прекоммент к форме поставить нельзя. Что делать?
3. Нарт. сказ. нет в списке ref, тоже забыла, что мы делаем в таком случае... Пока записала в ref текстом

**Comment by solanum-tuberosum** [comment]
А здесь тоже посткоммент хочет быть после перевода,а не до (слово ḱesena)
![image](https://user-images.githubusercontent.com/13102259/75096958-07b60680-55b6-11ea-8b40-e679f2410542.png)

**Comment by obelyaev** [comment]
>     1. кирг. написала в прекоммент к циту, это так и надо?
Нет, "кирг." есть в списке языков. Отмечайте в таких случаях два языка: "тюрк." и "кирг." Аналогично "анат.". Скобок не будет, но это не страшно. Можно было бы считать это диалектом, но на мой взгляд, если язык есть в списке языков, неправильно его ещё раз добавлять и в диалекты тоже.
 
>     2. вряд ли же "анат" относится к плащу? Наверно, это анат. панцирь? Там опечатка и это посткоммент к форме? Но посткоммент вылезает между словом и переводом... А прекоммент к форме поставить нельзя. Что делать?

Нет, он относится именно к плащу. Абаев приводит два слова: кирг. ХХХ "панцирь" и анат. ХХХ "плащ". Просто он иногда форматирует разные тюркские как отдельные языки, а иногда — как типа диалекты при одном "тюрк.".

> 
>     3. Нарт. сказ. нет в списке ref, тоже забыла, что мы делаем в таком случае... Пока записала в ref текстом

Текстом пишем. Когда много накопится, можно будет заблокировать и всё внести. Но вообще "Нарт. сказ." есть в списке осетинских источников, т.е. это не совсем уж неизвестная вещь.

**Comment by obelyaev** [comment]
> А здесь тоже посткоммент хочет быть после перевода,а не до (слово ḱesena)
> ![image](https://user-images.githubusercontent.com/13102259/75096958-07b60680-55b6-11ea-8b40-e679f2410542.png)

Мне кажется, в данном случае не страшно, если он будет после слова. А Вы бы как предложили решить проблему посткомментов в разных местах? Может, наоборот, сделать их всегда после перевода? От этого получится что-то плохое? У меня судя по Вашим Issues возникает ощущение, что посткоммент после перевода встречается чаще, чем посткоммент между словом и переводом. Это так?

**Comment by obelyaev** [comment]
Знаете что. Я что-то вошёл во вкус. Не пуститься ли нам во все тяжкие и не сделать ли перевод не атрибутом, а элементом? Заодно и текст формы перевести в разряд отдельных элементов. Так же, как я это сделал для Example. Тогда мы убиваем сразу двух зайцев: во-первых, решаем этот Issue, позволяя добавлять комментарии к отдельным переводам или к самому тексту формы; во-вторых, не будет уродливых форм с одним переводом, без собственно текста.

Единственный минус — это несколько усложнённый вид этих элементов в этимологиях и т.д. (очень длинные цепочки тэгов), но у нас там и так всё довольно плохо, так что вряд ли можно сделать хуже :))

**Comment by solanum-tuberosum** [comment]
> Знаете что. Я что-то вошёл во вкус. Не пуститься ли нам во все тяжкие и не сделать ли перевод не атрибутом, а элементом? Заодно и текст формы перевести в разряд отдельных элементов. Так же, как я это сделал для Example. Тогда мы убиваем сразу двух зайцев: во-первых, решаем этот Issue, позволяя добавлять комментарии к отдельным переводам или к самому тексту формы; во-вторых, не будет уродливых форм с одним переводом, без собственно текста.
> 
> Единственный минус — это несколько усложнённый вид этих элементов в этимологиях и т.д. (очень длинные цепочки тэгов), но у нас там и так всё довольно плохо, так что вряд ли можно сделать хуже :))

Ой, я что-то пока не до конца поняла, как именно это будет выглядеть. По логике же у нас в каждом ците должен быть текст, а к нему уже можно добавлять переводы. но при этом в ците нужно мочь добавлять несколько разных  текстов (т.е. это же не совсем как example, потому что в каждом примере у нас только один, собственно, пример и только один перевод. а в цитах может быть несколько словоформ и у каждой может быть несколько переводов) или это нестрашно, и все будет нормально в варианте, который вы предлагаете?

И вот про это, кстати: 
> > А здесь тоже посткоммент хочет быть после перевода,а не до (слово ḱesena)
> > ![image](https://user-images.githubusercontent.com/13102259/75096958-07b60680-55b6-11ea-8b40-e679f2410542.png)
> 
> Мне кажется, в данном случае не страшно, если он будет после слова. А Вы бы как предложили решить проблему посткомментов в разных местах? Может, наоборот, сделать их всегда после перевода? От этого получится что-то плохое? У меня судя по Вашим Issues возникает ощущение, что посткоммент после перевода встречается чаще, чем посткоммент между словом и переводом. Это так?

написала в закрытом ишью (https://github.com/abaevdict/abaevdict/issues/111#issuecomment-654079859) вот здесь, например, нужен прекоммент (чтобы показывался после слова до перевода) 
а во всех остальных случаях нужен собственно посткоммент (чтобы показывался после перевода)

**Comment by obelyaev** [comment]
> > Знаете что. Я что-то вошёл во вкус. Не пуститься ли нам во все тяжкие и не сделать ли перевод не атрибутом, а элементом? Заодно и текст формы перевести в разряд отдельных элементов. Так же, как я это сделал для Example. Тогда мы убиваем сразу двух зайцев: во-первых, решаем этот Issue, позволяя добавлять комментарии к отдельным переводам или к самому тексту формы; во-вторых, не будет уродливых форм с одним переводом, без собственно текста.
> > Единственный минус — это несколько усложнённый вид этих элементов в этимологиях и т.д. (очень длинные цепочки тэгов), но у нас там и так всё довольно плохо, так что вряд ли можно сделать хуже :))
> 
> Ой, я что-то пока не до конца поняла, как именно это будет выглядеть. По логике же у нас в каждом ците должен быть текст, а к нему уже можно добавлять переводы. но при этом в ците нужно мочь добавлять несколько разных текстов (т.е. это же не совсем как example, потому что в каждом примере у нас только один, собственно, пример и только один перевод. а в цитах может быть несколько словоформ и у каждой может быть несколько переводов) или это нестрашно, и все будет нормально в варианте, который вы предлагаете?
> 
> И вот про это, кстати:
> 
> > > А здесь тоже посткоммент хочет быть после перевода,а не до (слово ḱesena)
> > > ![image](https://user-images.githubusercontent.com/13102259/75096958-07b60680-55b6-11ea-8b40-e679f2410542.png)
> > 
> > 
> > Мне кажется, в данном случае не страшно, если он будет после слова. А Вы бы как предложили решить проблему посткомментов в разных местах? Может, наоборот, сделать их всегда после перевода? От этого получится что-то плохое? У меня судя по Вашим Issues возникает ощущение, что посткоммент после перевода встречается чаще, чем посткоммент между словом и переводом. Это так?
> 
> написала в закрытом ишью ([#111 (comment)](https://github.com/abaevdict/abaevdict/issues/111#issuecomment-654079859)) вот здесь, например, нужен прекоммент (чтобы показывался после слова до перевода)
> а во всех остальных случаях нужен собственно посткоммент (чтобы показывался после перевода)

Будет что-то вроде:

`<cit><form><PreComment>(из верхнестаровогул.)</PreComment><text>ĥʁwj<PostComment>(не может быть отделено от ħrjd id.)</PostComment></text><tr>небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг<PostComment>перевод неясен</PostComment></tr><tr>маленькая чашка</tr><PostComment>слово не встречается в современых словарях</PostComment></form></cit>`

и выглядеть должно как:

(из верхнестаровогул.) _ĥʁwj_ (не может быть отделено от _ħrjd_ id.) 'небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг' (перевод неясен), 'маленькая чашка' (слово не встречается в современых словарях)

Т.е. здесь есть прекоммент к форме, отдельный посткоммент к ТЕКСТУ формы, посткоммент к первому переводу и посткоммент ко всей форме.

**Comment by solanum-tuberosum** [comment]
> Будет что-то вроде:
> 
> `<cit><form><PreComment>(из верхнестаровогул.)</PreComment><text>ĥʁwj<PostComment>(не может быть отделено от ħrjd id.)</PostComment></text><tr>небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг<PostComment>перевод неясен</PostComment></tr><tr>маленькая чашка</tr><PostComment>слово не встречается в современых словарях</PostComment></form></cit>`
> 
> и выглядеть должно как:
> 
> (из верхнестаровогул.) _ĥʁwj_ (не может быть отделено от _ħrjd_ id.) 'небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг' (перевод неясен), 'маленькая чашка' (слово не встречается в современых словарях)
> 
> Т.е. здесь есть прекоммент к форме, отдельный посткоммент к ТЕКСТУ формы, посткоммент к первому переводу и посткоммент ко всей форме.

т.е. у cit'а все еще есть form, в которой может быть только один text, но к которой можно добавить много переводов tr? Вроде звучит неплохо

**Comment by obelyaev** [comment]
> > Будет что-то вроде:
> > `<cit><form><PreComment>(из верхнестаровогул.)</PreComment><text>ĥʁwj<PostComment>(не может быть отделено от ħrjd id.)</PostComment></text><tr>небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг<PostComment>перевод неясен</PostComment></tr><tr>маленькая чашка</tr><PostComment>слово не встречается в современых словарях</PostComment></form></cit>`
> > и выглядеть должно как:
> > (из верхнестаровогул.) _ĥʁwj_ (не может быть отделено от _ħrjd_ id.) 'небольшой сосуд для хранения крошек, оставшихся после вкушения хлеба в ночь со среды на четверг' (перевод неясен), 'маленькая чашка' (слово не встречается в современых словарях)
> > Т.е. здесь есть прекоммент к форме, отдельный посткоммент к ТЕКСТУ формы, посткоммент к первому переводу и посткоммент ко всей форме.
> 
> т.е. у cit'а все еще есть form, в которой может быть только один text, но к которой можно добавить много переводов tr? Вроде звучит неплохо

Да, идея именно в этом, чтобы можно было комментарии делать к отдельным элементам. Единственная проблема — получается громоздко.

Кто-нибудь помнит, для чего вообще у form есть поле def? Для переводов без кавычек? А оно вообще хоть когда-нибудь используется?

**Comment by solanum-tuberosum** [comment]
мне кажется да, для переводов без кавычек. но мб просто сделать какую-то штуку для tr типа no_quotes?)) по ощущениям вроде def очень редко нужен

**Comment by solanum-tuberosum** [comment]
вот как раз встретился деф
![image](https://user-images.githubusercontent.com/13102259/98215947-7b881b80-1f59-11eb-8991-9f61a944e8ca.png)
